### PR TITLE
Converted to Python3

### PR DESCRIPTION
**Pull request by amunozj**
- Added parentheses to print
- Changed open statements from open(xxxxx, "rb") to open(xxxxx, "r")

### CODE REVIEW

File: dataset_models/sdo/aia.py [hunk 1]
```diff
@@ -15,7 +17,7 @@ class AIA(dataset_models.dataset.Dataset):
 
     def __init__(self,
                  samples_per_step=32,
-                 dependent_variable="flux delta",
+                 dependent_variable="forecast",
```
Comment (smcgregor) [inline dataset_models/sdo/aia.py]: Normally you would not want this line in a pull request dedicated to only making the code Python3 compliant.

File: dataset_models/sdo/aia.py [hunk 2]
```diff
@@ -70,12 +72,13 @@ def __init__(self,
...
+        with open(self.y_filepath, "r") as f:
             f.readline()
             for line in f:
-                split_y = line.split(",")
-                cur_y = float(split_y[1])
-                self.y_dict[split_y[0]] = cur_y
+                split_y = line.split(',')
+                #cur_y = float(split_y[1])
+                cur_y = float(split_y[3])
+                self.y_dict[split_y[0]] = cur_y*1e6
```
Comment (smcgregor) [inline dataset_models/sdo/aia.py]: What is going on here? This looks like a semantic change rather than a syntactic one.

File: network_models/training_callbacks.py [hunk 3]
```diff
@@ -28,9 +28,9 @@ def on_train_end(self, logs={}):
 
         with open(self.filepath, "ab") as out:
             for loss in self.losses:
-                out.write(str(loss[0]))
+                out.write((loss[0]))
                 out.write(",")
-                out.write(str(loss[1]))
+                out.write((loss[1]))
```
Comment (smcgregor) [inline network_models/training_callbacks.py]: Extra parens.

File: network_models/training_callbacks.py [hunk 4]
```diff
@@ -28,9 +28,9 @@ def on_train_end(self, logs={}):
 
         with open(self.filepath, "ab") as out:
             for loss in self.losses:
-                out.write(str(loss[0]))
+                out.write((loss[0]))
```
Comment (smcgregor) [inline network_models/training_callbacks.py]: Extra parens.

File: network_models/xray_flux_forecast/starting_point/network.py [hunk 5]
```diff
@@ -18,11 +18,16 @@
 # Optimizer used in training
 from keras.optimizers import adam
 
+# Uncomment to force training to take place on the CPU
+#import os
```
Comment (smcgregor) [inline network_models/xray_flux_forecast/starting_point/network.py]: There is a config parameter for this that will make the experiment.py target the CPU. You can remove this.

File: tools/xray_flux_forecast/v4_find_ys.py [hunk 6]
```diff
@@ -24,8 +24,8 @@
 import os
 import os.path
 
-filePath = "dataset_models/sdo/"
-dependent_variable_path = "dataset_models/sdo/y/"
+filePath = "/home/amunozj/FlareAIA/"
```
Comment (smcgregor) [inline tools/xray_flux_forecast/v4_find_ys.py]: These should stay as relative paths.

File: dataset_models/sdo/aia.py [hunk 7]
```diff
@@ -402,9 +406,9 @@ def _get_prior_x_filename(self, filename, timestep):
         """
         identifier = self._get_prior_timestep_string(filename, timestep)
         if filename[:5] == "noflr":
-            suffix = "_8chnls_1024_0" + str(12 * timestep) + "m.fthr"
+            suffix = "_8chnls_1024_0" + str(12 * timestep) + "h.fthr"
```
Comment (smcgregor) [inline dataset_models/sdo/aia.py:409]: Should I be worried about these changes? What are the implications? This should really be in a separate pull request. (same below)

File: dataset_models/sdo/aia.py [hunk 8]
```diff
@@ -460,12 +465,12 @@ def _get_aia_image(self, filename, directory, previous=0):
         assert previous >= 0, "previous should be a non-negative integer, it is currently " + previous
         assert previous < 5, "previous should not be a very large integer, it is currently " + previous
         if previous == 0:
-            data = feather.read_dataframe(directory + filename)
-            return data.values
+            data = feather.read_dataframe(directory + filename).values.reshape(1024,1024,8)
```
Comment (smcgregor) [inline dataset_models/sdo/aia.py:468]: Why is this reshape now necessary? Same below.

**Review by smcgregor — CHANGES_REQUESTED** [review]
There are a few stylistic things that need to be changed, as well as a few semantic changes that either need to be removed or justified.

File: dataset_models/sdo/aia.py (hunk 1 above)
Comment (amunozj) [reply dataset_models/sdo/aia.py]: Removed this change

File: dataset_models/sdo/aia.py (hunk 2 above)
Comment (amunozj) [reply dataset_models/sdo/aia.py]: Removed this change

File: dataset_models/sdo/aia.py (hunk 7 above)
Comment (amunozj) [reply dataset_models/sdo/aia.py:409]: It is nothing critical, just a naming change that I'm currently using.  Can we leave it?  The download scripts (once they are committed) will be consistent

File: dataset_models/sdo/aia.py (hunk 8 above)
Comment (amunozj) [reply dataset_models/sdo/aia.py:468]: It was sometime ago, but I believe it was necessary for it to work.  I'm not sure why.  For some reason in python3 the shape was all wrong.

Same below.

File: network_models/training_callbacks.py (hunk 4 above)
Comment (amunozj) [reply network_models/training_callbacks.py]: fixed.  Removing str was necessary, though

File: network_models/training_callbacks.py (hunk 3 above)
Comment (amunozj) [reply network_models/training_callbacks.py]: fixed

File: network_models/xray_flux_forecast/starting_point/network.py (hunk 5 above)
Comment (amunozj) [reply network_models/xray_flux_forecast/starting_point/network.py]: removed

File: tools/xray_flux_forecast/v4_find_ys.py (hunk 6 above)
Comment (amunozj) [reply tools/xray_flux_forecast/v4_find_ys.py]: does this mean that we need to put the data inside the folder structure?  that is not possible in my case

File: dataset_models/sdo/aia.py (hunk 7 above)
Comment (smcgregor) [reply dataset_models/sdo/aia.py:409]: This is a better time to change this than any in the future. I assume it requires re-downloading data? This is worth putting in the summary at the top of the pull request.

File: tools/xray_flux_forecast/v4_find_ys.py (hunk 6 above)
Comment (smcgregor) [reply tools/xray_flux_forecast/v4_find_ys.py]: Maybe we need to have this file reference the global config.yml?

Comment (smcgregor) [reply tools/xray_flux_forecast/v4_find_ys.py]: Then you could change the config once and not commit the changes.

File: dataset_models/sdo/aia.py (hunk 8 above)
Comment (smcgregor) [reply dataset_models/sdo/aia.py:468]: I think we need to visualize this to make sure the new way is good. I am worried about introducing some weird non-coherent expression of the data.

File: dataset_models/sdo/aia.py (hunk 7 above)
Comment (amunozj) [reply dataset_models/sdo/aia.py:409]: I will put it then.  This was a change that happened on the last week of the FDL (blush)

File: tools/xray_flux_forecast/v4_find_ys.py (hunk 6 above)
Comment (amunozj) [reply tools/xray_flux_forecast/v4_find_ys.py]: that's the way it was before, I believe.

Comment (smcgregor) [reply tools/xray_flux_forecast/v4_find_ys.py]: Click through to this comment and you'll see how it was before.

File: network_models/xray_flux_forecast/starting_point/network.py [hunk 9]
```diff
@@ -84,16 +79,24 @@
     x = concatenate(input_images)
 else:
     x = input_images[0]
-x = Conv2D(1, (1,1), strides=(1,1), padding='same', activation="relu")(x)
-x = MaxPooling2D(pool_size=(1024, 1024), strides=(1,1), padding='valid')(x)
+    
+x = Conv2D(16, kernel_size=16, padding = 'same', activation='relu', use_bias=False, kernel_initializer = 'lecun_uniform')(x)
+x = Conv2D(16, kernel_size=16, padding = 'same', activation='relu', use_bias=False, kernel_initializer = 'lecun_uniform')(x)
```
Comment (smcgregor) [inline network_models/xray_flux_forecast/starting_point/network.py:84]: It looks like you brought in a bunch of network changes accidentally. If you are doing this from the command line, try `git add -p network_models/xray_flux_forecast/starting_point/network.py` next time.

**Comment by smcgregor** [comment]
@amunozj Updates look good. One small comment that isn't a show stopper. I am ready to merge if you are good.

**Comment by smcgregor** [comment]
Tagging the issue for future reference #31

**Review by smcgregor — APPROVED** [review]
